raspberry.rb

```diff
@@ -0,0 +1,124 @@
+class Raspberry
+  @@states = %w[absence bloom green red]
+
+  def initialize(index)
+    @index = index
+    @state = @@states.first
+  end
+
+  def grow!
+    i = @@states.index(@state)
+    if i == 3
+      puts "Малина уже созрела!"
+    else
+      @state = @@states[i + 1]
+    end
+  end
+
+  def ripe?
+    @state.eql?(@@states[3])
+  end
+end
+
+class RaspberryBush
+
+  def initialize(raspberries_amount)
+    @raspberry_bash = Array.new(raspberries_amount)
+    (0..raspberries_amount).each { |i|
+      @raspberry_bash[i] = Raspberry.new(i)
+    }
+  end
+
+  def grow_all!
+    counter = 0
+    while counter < @raspberry_bash.length do
+      @raspberry_bash[counter].grow!
+      counter += 1
+    end
+  end
+
+  def ripe_all?
+    counter = 0
+    flag = true
+    while counter < @raspberry_bash.length do
+      flag = @raspberry_bash[counter].ripe?
+      counter += 1
+      unless flag
+        break
+      end
+    end
+    flag
+  end
+
+  def give_away_all!
+    @raspberry_bash.clear
+  end
+end
+
+class Human
+  private(attr_reader :plant)
```

Comment:
It is better to write all public methods first, then private ones:
```suggestion
  private
  
  attr_reader :plant
```